src/ts/contentScripts/live-preview/index.tsx

```diff
@@ -0,0 +1,169 @@
+import {Feature, Settings} from '../../utils/settings'
+import {Navigation} from '../../roam/navigation'
+import {browser} from 'webextension-polyfill-ts'
+import {createPopper, Instance} from '@popperjs/core'
+
+export const config: Feature = {
+    id: 'live_preview',
+    name: 'Live Preview',
+    defaultIsActive: false,
+    warningTooltip: 'Experimental feature; Large databases might see performance issues.',
+    settings: [],
+}
+
+Settings.isActive('live_preview').then(active => {
+    if (active) {
+        enableLivePreview()
+    }
+})
+
+browser.runtime.onMessage.addListener(async message => {
+    if (message?.featureId === 'live_preview' && message?.type === 'toggle') {
+        if (message.value) {
+            enableLivePreview()
+        } else {
+            removeLivePreview()
+        }
+    }
+})
```

Comment:
@Stvad how to know when settings are updated? I thought we need to use onMessage for that?

Reply:
yeah, but it's structurally simpler, so you just need to add another thing to trigger here https://github.com/roam-unofficial/roam-toolkit/blob/master/src/ts/contentScripts/dispatcher/index.ts#L18

Reply:
I don't seem to be able to get that working -- the actions in that dispatcher do not seem to be triggered from anywhere. All keyboard shortcuts, for eg, are triggered from their own `onPress`. 

Also I see that `updateShortcuts` will be called on every toggle/update, which might not be needed for live preview.

Reply:
I can add a seperate trigger for this,  but it doesn't seem right --- we will need to couple all the code for preview to dispatcher -- even now, the dispatcher has a lot of coupling, which makes it difficult to modify

Reply:
using `settings-updated` for now -- it seems like something needs to change here.
`settings-updated` does not tell us which setting is updated, so it just calls `updateShortcuts` even any setting is toggled. 

I am checking for live-preview setting when any setting is updated, and taking actions accordingly.

Also related: CSS toggle does not really work (even in the published extension). This is because on any toggle, we only dispatch `settings-updated`,which does not impact the CSS.

I have ideas around how to fix this flow to be a little decouples from each other -- that way settings don't know about each other, and it would fix some of these issues. Let's keep them for a separate change @Stvad

Reply:
agree for making settings changes elsewhere.  
For `settings-updated` - it's true that it's coarse (i.e. does not tell you what setting changed etc).  
For the use cases we see I think it's ok though - it's simple. And even though it may trigger reloading of things more often on settings change - that's fine because settings change is a very rare event. 

Replied to you on slack for things in dispatcher not being triggered and did some cleanup

Reply:
for better interface for dispatcher, I imagine we can invert the flow of the dependencies, and make it implement observer. So it'd be an object/class that would have interface like

```javascript
onSettingsUpdated(function)
onSettingUpdated(name, function) //when we'd need a more granular things

...etc
```

Reply:
Yes, this would be a good pattern too. Similar to onMessage, but less explicit